File: examples/trixi_controller_simple.c
#include <stdio.h>

#include <trixi.h>

int main ( int argc, char *argv[] ) {

    if ( argc < 2 ) {
        fprintf(stderr, "ERROR: missing arguments: PROJECT_DIR LIBELIXIR_PATH\n\n");
        fprintf(stderr, "usage: %s PROJECT_DIR LIBELIXIR_PATH\n", argv[0]);
        return 2;
    } else if ( argc < 3 ) {
        fprintf(stderr, "ERROR: missing argument: LIBELIXIR_PATH\n\n");
        fprintf(stderr, "usage: %s PROJECT_DIR LIBELIXIR_PATH\n", argv[0]);
        return 2;
    }

    // Initialize Trixi
    printf("\n*** Trixi controller ***   Initialize Trixi\n");
    trixi_initialize( argv[1], NULL );

    // Set up the Trixi simulation
    // We get a handle to use subsequently
    printf("\n*** Trixi controller ***   Set up Trixi simulation\n");
    int handle = trixi_initialize_simulation( argv[2] );

    // Main loop
    printf("\n*** Trixi controller ***   Entering main loop\n");
    while ( !trixi_is_finished( handle ) ) {

        trixi_step( handle );
    }

    // Finalize Trixi simulation
    printf("\n*** Trixi controller ***   Finalize Trixi simulation\n");
    trixi_finalize_simulation( handle );

    // Finalize Trixi
    printf("\n*** Trixi controller ***   Finalize Trixi\n");
    trixi_finalize();

    return 0;
}

File: examples/trixi_controller_simple.jl
using LibTrixi

if length(ARGS) < 1
    error("ERROR: missing argument LIBELIXIR_PATH")
end

println("*** Trixi controller ***   Set up Trixi simulation")
handle = trixi_initialize_simulation(ARGS[1])

println("*** Trixi controller ***   Current time step length: ", trixi_calculate_dt(handle))

println("*** Trixi controller ***   Entering main loop")

while trixi_is_finished(handle) == 0
    trixi_step(handle)
end

println("*** Trixi controller ***   Finalize Trixi simulation")
trixi_finalize_simulation(handle)
